### ISSUE TITLE
Upgrade to Kotlin 1.6.0

### ISSUE THREAD

**Issue by snicoll** [spring-projects/spring-boot#28698]
Supersedes #28577

**Comment by hennr** [comment]
Hi @snicoll ,
sorry if I am getting something wrong here, but I just saw that the spring-boot-dependencies artifact on mvn central still uses kotlin.version 1.5.31:
https://search.maven.org/artifact/org.springframework.boot/spring-boot-dependencies/2.6.0/pom

The SB 2.6.0 changelog states that kotlin 1.6.0 is used now though.

Is this a bug or don't I get something correctly here?

**Comment by bclozel** [comment]
@hennr [This file](https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/2.6.0/spring-boot-dependencies-2.6.0.pom) shows the following line for me:

```xml
<kotlin.version>1.6.0</kotlin.version>
</properties>
<!-- ... -->
<dependency>
  <groupId>org.jetbrains.kotlin</groupId>
  <artifactId>kotlin-bom</artifactId>
  <version>${kotlin.version}</version>
  <type>pom</type>
  <scope>import</scope>
</dependency>
```

**Comment by hennr** [comment]
It does indeed!
Strange, so the maven central web-ui seems broken in some way then:

https://search.maven.org/artifact/org.springframework.boot/spring-boot-dependencies/2.6.0/pom
![grafik](https://user-images.githubusercontent.com/677418/142843398-96ed8187-1d56-45b3-b2ea-893ac44d031b.png)

**Comment by wilkinsona** [comment]
Indeed. When I first go to the page, no source is shown for the pom. If I switch to 2.5.7, it shows the source for 2.5.7's pom. Switching back to 2.6.0 then continues to show the source for 2.5.7. Can you please report this to Sonatype?

**Comment by hennr** [comment]
@wilkinsona , sure, here you go
https://github.com/sonatype-nexus-community/search-maven-org/issues/159

One more question though: Could you check if the artifact upload was completely successful on your end?

I have the exact same behaviour that you described, but what's bugging me is that a click on browse in upper right corner succeeds but a click on the download button fails with a 404.

**Comment by wilkinsona** [comment]
The two are different URLs:

- Download: https://search.maven.org/remotecontent?filepath=org/springframework/boot/spring-boot-dependencies/2.6.0/spring-boot-dependencies-2.6.0.pom
- Browse: https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/2.6.0/

I would avoid using search.maven.org when trying to check Maven Central's contents. Going directly to repo1.maven.org is, IMO, the right thing to do as it eliminates the possibility of search.maven.org causing the problem either due to a bug or because its index is stale.

**Comment by hennr** [comment]
It's been a caching problem on search.maven.org and it's shown correctly now, see https://github.com/sonatype-nexus-community/search-maven-org/issues/159 for details.
Thanks for your time and input on this!
